docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,255 @@
+```
+
+A common use case could be to set the "admin" as the caller uploading the
+package.
+
+```go
+import (
+    "std"
+    "time"
+)
+
+var (
+    created time.Time
+    admin   std.Address
+    list    []string
+)
+
+func init() {
+    created = time.Now()
+    admin = std.GetOrigCaller()
+    list = append(list, "foo", "bar")
+}
+```
+
+In essence, `init()` in Gno is your go-to function for setting up and
+registering realms. It's a powerful tool that helps keep your realms organized
+and properly configured from the get-go.
+
+## Gno Good Practices
+
+### Design Your Realm as a Public API
+
+In Go, all your packages, including your dependencies, are typically treated as
+part of your safe zone, similar to a secure perimeter. The boundary is drawn
+between your program and the rest of the world, which means you secure the API
+itself, potentially with authentication middlewares.
+
+However, in Gno, your package is the public API. It's exposed to the outside
+world and can be accessed by other realms. Therefore, it's crucial to design
+your realm with the same level of care and security considerations as you would
+a public API.
+
+One approach is to simulate a secure perimeter within your realm by having
+private functions for the logic and then writing your API layer by adding some
+front-facing API with authentication. This way, you can control access to your
+realm's functionality and ensure that only authorized callers can execute
+certain operations.
+
+```go
+import "std"
+
+func PublicMethod(nb int) {
+    caller := std.GetOrigCaller()
+    privateMethod(caller, nb)
+}
+
+func privateMethod(caller std.Address, nb int) { /* ... */ }
+```
+
+In this example, `PublicMethod` is a public function that can be called by other
+realms. It retrieves the caller's address using `std.GetOrigCaller()`, and then
+passes it to `privateMethod`, which is a private function that performs the
+actual logic. This way, `privateMethod` can only be called from within the
+realm, and it can use the caller's address for authentication or authorization
+checks.
+
+### Construct "Safe" Objects
+
+A safe object in Gno is an object that is designed to be tamper-proof and
+secure. It's created with the intent of preventing unauthorized access and
+modifications. This follows the same principle of making a package an API, but
+for a Go object that can be directly referenced by other realms.
+
+The goal is to create an object which, once instantiated, can be linked and its
+pointer can be "stored" by other realms without issue, because it protects its
+usage completely.
+
+```go
+type MySafeStruct {
+    counter nb
+    admin std.Address
+}
+
+func NewSafeStruct() *MySafeStruct {
+    caller := std.GetOrigCaller()
+    return &MySafeStruct{
+        counter: 0,
+        admin: caller,
+    }
+}
+
+func (s *MySafeStruct) Counter() int { return s.counter }
+func (s *MySafeStruct) Inc() {
+    caller := std.GetOrigCaller()
+    if caller != s.admin {
```

Comment:
Something like this could be used in the above example for access control, or consider combining sections to avoid redundancy

Reply:
A secure struct always involves access control. Therefore, we should address access control in that section.

Additionally, we should have a separate access control section, specifically discussing access control at the contract level rather than the object level.

I will create a new section on contract-level access control.

Reply:
e5d95a2c